tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -354,71 +345,73 @@ Operation* convertUnpackOp(PatternRewriter& rewriter, Operation* op,
 
   // Combine the sequence of tosa.slice() ops into a list
   // using the IdentityN operator.
-  return rewriter.create<tosa::IdentityNOp>(
-      op->getLoc(), ArrayRef<Type>(outs_type_vec), results_vec);
+  return rewriter
+      .create<tosa::IdentityNOp>(op->getLoc(), ArrayRef<Type>(outs_type_vec),
+                                 results_vec)
+      .getResults();
 }
 
 // Lowers the Select operator to TOSA.
-Operation* convertSelectOp(PatternRewriter& rewriter, Operation* op,
-                           Value result_value, Value condition_value,
-                           Value x_value, Value y_value) {
+llvm::Optional<Value> convertSelectOp(PatternRewriter& rewriter, Operation* op,
+                                      Value result_value, Value condition_value,
+                                      Value x_value, Value y_value) {
   auto result_type = result_value.getType().dyn_cast<RankedTensorType>();
   auto condition_type = condition_value.getType().dyn_cast<RankedTensorType>();
   auto x_type = x_value.getType().dyn_cast<RankedTensorType>();
   auto y_type = y_value.getType().dyn_cast<RankedTensorType>();
 
-  Operation* result_op = nullptr;
-
   if (!result_type || !condition_type || !x_type || !y_type) {
     op->emitOpError("Select: failed ranked tensor type check");
-    return nullptr;
+    return llvm::None;
   }
 
   // First check whether we need to reshape the condition to match
   // the same rank as the then/else clauses.
   if (result_type.getRank() == condition_type.getRank()) {
     // Nothing to reshape.
-    result_op = rewriter.create<tosa::SelectOp>(
-        op->getLoc(), result_type, condition_value, x_value, y_value);
-  } else {
-    // Need to reshape the condition.
-    SmallVector<int64_t, 8> new_cond_dims;
-    for (int i = 0; i < (result_type.getRank() - condition_type.getRank());
-         i++) {
-      new_cond_dims.push_back(1);
-    }
-    for (int i = 0; i < condition_type.getRank(); i++) {
-      new_cond_dims.push_back(condition_type.getShape()[i]);
-    }
+    return rewriter
+        .create<tosa::SelectOp>(op->getLoc(), result_type, condition_value,
+                                x_value, y_value)
+        .getResult();
+  }
 
-    auto reshape_op = rewriter.create<tosa::ReshapeOp>(
-        op->getLoc(),
-        RankedTensorType::get(ArrayRef<int64_t>(new_cond_dims),
-                              condition_type.getElementType()),
-        condition_value, rewriter.getI64ArrayAttr(new_cond_dims));
+  // Need to reshape the condition.
+  SmallVector<int64_t, 8> new_cond_dims;
+  for (int i = 0; i < (result_type.getRank() - condition_type.getRank()); i++) {
+    new_cond_dims.push_back(1);
+  }
```

Comment:
Isn't this equivalent to 
```
  SmallVector<int64_t, 8> new_cond_dims(result_type.getRank() - condition_type.getRank()), 1);
```
?

Reply:
Thanks for the hint!  I wasn't aware of this SmallVector constructor.